### PR TITLE
[risk=low][no ticket] Add a little more null-safety to API createRuntime()

### PR DESCRIPTION
**Pull request by jmthibault79**
I happened to notice some gaps in NPE prevention while looking at this code.  No expected changes except for an edge case: if a GceWithPDConfig is passed to CreateRuntime without a PD Request (an invalid scenario), it will now throw a BadRequestException instead of a NullPointerException.  I'm not aware of any UI paths that would lead to this situation. 

Tested locally: createRuntime continues to work 

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java [hunk 1]
```diff
@@ -160,50 +161,52 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
...
+            // Find active disks for runtime VM. Block user from creating new disk.
+            throw new BadRequestException(
+                String.format(
+                    "Can not create new runtime with new PD if user has active runtime PD. Existing disks: %s",
+                    PersistentDiskUtils.prettyPrintDiskNames(runtimeDisks)));
+          }
+          persistentDiskRequest.name(userProvider.get().generatePDName());
         }
-        persistentDiskRequest.name(userProvider.get().generatePDName());
+        var labels = persistentDiskRequest.getLabels();
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:202]: this was not null-safe before

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java [hunk 2]
```diff
@@ -160,16 +161,27 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
...
+          "Exactly one of GceConfig or DataprocConfig or GceWithPdConfig must be provided");
     }
 
     DbWorkspace dbWorkspace = workspaceService.lookupWorkspaceByNamespace(workspaceNamespace);
 
     GceWithPdConfig gceWithPdConfig = runtime.getGceWithPdConfig();
     if (gceWithPdConfig != null) {
       PersistentDiskRequest persistentDiskRequest = gceWithPdConfig.getPersistentDisk();
-      if (persistentDiskRequest != null && Strings.isNullOrEmpty(persistentDiskRequest.getName())) {
+      if (persistentDiskRequest == null) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:180]: new check: this condition is not valid.  We also assume that this value is not null further down, so this also prevents NPE

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java [hunk 3]
```diff
@@ -160,16 +161,27 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
...
+        throw new BadRequestException("GceWithPdConfig must contain a PersistentDiskRequest");
+      }
+
+      if (Strings.isNullOrEmpty(persistentDiskRequest.getName())) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:184]: split from the previous condiftion

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java [hunk 4]
```diff
@@ -190,21 +202,14 @@ public ResponseEntity<EmptyResponse> createRuntime(String workspaceNamespace, Ru
         }
         persistentDiskRequest.name(userProvider.get().generatePDName());
       }
+
       var labels = persistentDiskRequest.getLabels();
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:206]: Previously, `persistentDiskRequest` was not guaranteed to be defined

File: api/src/test/java/org/pmiops/workbench/api/RuntimeControllerTest.java [hunk 5]
```diff
@@ -954,6 +954,16 @@ public void testCreateRuntime_gceWithPD() throws ApiException {
     assertThat(createLeonardoGceWithPdConfig.getZone()).isEqualTo("us-central-1");
   }
 
+  @Test
+  public void testCreateRuntime_gceWithPD_noPDRequest() {
+    // don't add the required persistentDisk field
+    var gceWithPdConfig_lacksPD =
+        new Runtime().gceWithPdConfig(new GceWithPdConfig().machineType("standard"));
+    assertThrows(
+        BadRequestException.class,
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/RuntimeControllerTest.java:963]: before this change, this would result in NPE

**Review by freemabd — APPROVED** [review]
LGTM

**Pull request merged**